paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/NakadiEventErrorHandlerConfiguration.java

```diff
@@ -31,4 +34,25 @@ public EventErrorHandlerList eventErrorHandlerList() {
             return new EventErrorHandlerList();
         }
     }
+
+    @Bean
+    public FailedEventSourceMap failedEventSourceMap() {
+
+        final Map<String, FailedEventSource> failedEventSourceMap = applicationContext.getBeansOfType(
+                FailedEventSource.class);
+
+        if (null != failedEventSourceMap) {
+            return new FailedEventSourceMap(failedEventSourceMap.entrySet().stream().collect(Collectors.toMap(k ->
+                                k.getValue().getEventSourceName(),
+                            Map.Entry::getValue)));
+        } else {
+            return new FailedEventSourceMap();
+        }
+    }
+
+    @Bean
+    public FailedEventReplayer failedEventHandler(final EventReceiverRegistry eventReceiverRegistry,
+            final FailedEventSourceMap failedEventSourceMap) {
+        return new FailedEventReplayer(eventReceiverRegistry, failedEventSourceMap, new ReplayHandler());
```

Comment:
As written before, better pass `List<FailedEventSource>` instead of `FailedEventSourceMap`.